application.go

```diff
@@ -103,28 +103,20 @@ func (a *Application) Run() error {
 
 	// Start event loop.
 	for {
+		// Do not poll events during suspend mode
+		a.suspendMutex.Lock()
 		a.Lock()
 		screen := a.screen
-		if a.suspended {
-			a.suspended = false // Clear previous suspended flag.
-		}
 		a.Unlock()
 		if screen == nil {
+			a.suspendMutex.Unlock()
 			break
 		}
 
 		// Wait for next event.
 		event := a.screen.PollEvent()
+		a.suspendMutex.Unlock()
```

Comment:
Doesn't it make sense to wrap the entire `for` block with the lock? It seems to me that the code after this `Unlock()` could also potentially interfere with suspended mode.

Reply:
No, in this case there will be a deadlock, if `Suspend()` will be called from `inputCapture()` handler (same goroutine)

Reply:
Right. That's true.